pkg/cmd/server/start/start_master.go

```diff
@@ -676,11 +676,16 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 	oc.RunServiceServingCertController(serviceServingCertClient)
 	oc.RunUnidlingController()
 
-	_, _, ingressIPClient, err := oc.GetServiceAccountClients(bootstrappolicy.InfraServiceIngressIPControllerServiceAccountName)
-	if err != nil {
-		glog.Fatalf("Could not get client: %v", err)
+	onBareMetal := (kc.CloudProvider == nil)
+	if onBareMetal {
+		// The ingress ip controller should only be run on baremetal. In a cloud environment, it will conflict with the
```

Comment:
How would that be possible?  In a cloud environment the service controller is active and relies on the same status field that the ingress controller uses.